### PR TITLE
Update distill_utils to support PP-MINILM

### PR DESCRIPTION
**Pull request by LiuChiachi**
### PR types

New features
### PR changes

APIs
### Description

Update distill_utils to support PP-MINILM

### CODE REVIEW

**Review by ZeyuChen — DISMISSED** [review]
LGTM

File: paddlenlp/transformers/distill_utils.py [hunk 1]
```diff
@@ -19,20 +19,163 @@
...
+
+        - `head-head loss` (Tensor):
+            head-head relation loss.
+
+        - `sample-sample loss` (Tensor):
+            sample-sample relation loss.
+
+    """
+    # Avoid NAN while training using AMP.
+    attn_mask /= 18000  # [bs, 1, 1, seq_len]
```
Comment (tianxin1860) [inline paddlenlp/transformers/distill_utils.py]: attn_mask 的 weight 统一改成 1E-4 吧，和原始 BERT 保持一致.

File: paddlenlp/transformers/distill_utils.py [hunk 2]
```diff
@@ -19,20 +19,163 @@
...
+
+    """
     # Initialize head_num
     if num_relation_heads > 0 and num_relation_heads != s.shape[1]:
         # s'shape: [bs, seq_len, head_num, head_dim]
         s = tensor.transpose(x=s, perm=[0, 2, 1, 3])
         # s'shape: [bs, seq_len, num_relation_heads, head_dim_new]
         s = tensor.reshape(x=s, shape=[0, 0, num_relation_heads, -1])
-        #s's shape: [bs, num_relation_heads, seq_len,, head_dim_new]
+        #'s shape: [bs, num_relation_heads, seq_len,, head_dim_new]
```
Comment (tianxin1860) [inline paddlenlp/transformers/distill_utils.py]: s'shape: [bs, num_relation_heads, seq_len,, head_dim_new]

File: paddlenlp/transformers/distill_utils.py [hunk 3]
```diff
@@ -19,20 +19,163 @@
...
+        scaled_dot_product_t2 += attn_mask_sample_sample
+        loss_sample_sample = loss_fct(
+            F.log_softmax(scaled_dot_product_s2),
+            F.softmax(scaled_dot_product_t2))
+    return (loss_token_token, loss_head_head, loss_sample_sample)
 
 
 def calc_minilm_loss(loss_fct, s, t, attn_mask, num_relation_heads=0):
+    """
+    Calculates loss for Q-Q, K-K, V-V relation from MiniLM.
```
Comment (tianxin1860) [inline paddlenlp/transformers/distill_utils.py]: MiniLMv2

File: paddlenlp/transformers/distill_utils.py [hunk 4]
```diff
@@ -19,20 +19,163 @@
...
+        s1 = tensor.transpose(x=s, perm=[0, 2, 1, 3])
+    if num_relation_heads > 0 and num_relation_heads != t.shape[1]:
+        t = tensor.transpose(x=t, perm=[0, 2, 1, 3])
+        t = tensor.reshape(x=t, shape=[0, 0, num_relation_heads, -1])
+        t1 = tensor.transpose(x=t, perm=[0, 2, 1, 3])
+
+    pad_seq_len = s.shape[1]
+    s_head_dim, t_head_dim = s.shape[3], t.shape[3]
+
+    if alpha + beta == 1.0:
```
Comment (tianxin1860) [inline paddlenlp/transformers/distill_utils.py:82]: 采用 alpha、beta 2 个 float 数值作为 bool 类型来控制 3 个 Loss 中的哪个 Loss 生效？直接用 Bool 类型？

Comment (tianxin1860) [reply paddlenlp/transformers/distill_utils.py:82]: 或者最终 return 的 Loss = alpha * head_head_loss + beta * sample_sample_loss + (1.0 - alpha - beta) * token_token_loss ?

**Review by tianxin1860 — COMMENTED** [review]
Leave some comments

File: paddlenlp/transformers/distill_utils.py (hunk 1 above)
Comment (tianxin1860) [reply paddlenlp/transformers/distill_utils.py]: 硬编码尽量避免.

File: paddlenlp/transformers/distill_utils.py (hunk 4 above)
Comment (LiuChiachi) [reply paddlenlp/transformers/distill_utils.py:82]: Done. Thanks!已经修改成
`return alpha * head_head_loss + beta * sample_sample_loss + (1.0 - alpha - beta) * token_token_loss `

File: paddlenlp/transformers/distill_utils.py [hunk 5]
```diff
@@ -19,20 +19,154 @@
...
+            x=s1, y=s1, transpose_y=True) / math.sqrt(s_head_dim)
+        del s1
+        scaled_dot_product_s1 += attn_mask
+        scaled_dot_product_t1 = tensor.matmul(
+            x=t1, y=t1, transpose_y=True) / math.sqrt(t_head_dim)
+        del t1
+        scaled_dot_product_t1 += attn_mask
+        loss_token_token = loss_fct(
+            F.log_softmax(scaled_dot_product_s1),
+            F.softmax(scaled_dot_product_t1))
```
Comment (guoshengCS) [inline paddlenlp/transformers/distill_utils.py:95]: 这里一个是log_softmax，一个是softmax，loss_fct应该会有限制，kl_div可以其他的估计不太直接，这里如果不改的话文档里最好补充说明下吧

File: paddlenlp/transformers/distill_utils.py [hunk 6]
```diff
@@ -19,20 +19,154 @@
 import paddle.nn.functional as F
 from paddle.nn import MultiHeadAttention, TransformerEncoderLayer, TransformerEncoder
 from paddle.fluid.data_feeder import convert_dtype
+
 from paddlenlp.utils.log import logger
-from paddlenlp.transformers import TinyBertForPretraining, TinyBertForSequenceClassification, BertForSequenceClassification
+from paddlenlp.transformers import ErnieForSequenceClassification
+from paddlenlp.transformers import TinyBertForPretraining, TinyBertForSequenceClassification
```
Comment (tianxin1860) [inline paddlenlp/transformers/distill_utils.py]: import TinyBertForSequenceClassification 之后没有用到？

**Review by tianxin1860 — COMMENTED** [review]
Leave a comment

File: paddlenlp/transformers/distill_utils.py (hunk 3 above)
Comment (LiuChiachi) [reply paddlenlp/transformers/distill_utils.py]: Done. Thanks!

File: paddlenlp/transformers/distill_utils.py (hunk 2 above)
Comment (LiuChiachi) [reply paddlenlp/transformers/distill_utils.py]: Done. Thanks!

File: paddlenlp/transformers/distill_utils.py (hunk 1 above)
Comment (LiuChiachi) [reply paddlenlp/transformers/distill_utils.py]: Done. Thanks!

File: paddlenlp/transformers/distill_utils.py (hunk 5 above)
Comment (LiuChiachi) [reply paddlenlp/transformers/distill_utils.py:95]: Done. Thanks!

File: paddlenlp/transformers/distill_utils.py (hunk 6 above)
Comment (LiuChiachi) [reply paddlenlp/transformers/distill_utils.py]: Done. Thanks!已经移除了没有用到的import

**Review by tianxin1860 — APPROVED** [review]
LGTM
